Testscripts/Linux/XDP-Action.sh

```diff
@@ -0,0 +1,138 @@
+function ping_with_tcpdump () {
+	local logSuffix=$1
+	LogMsg "Ips client $client server $server"
+	# Start tcpdump
+	LogMsg "Starting tcpdump on $client"
+	ssh ${client} "sh -c 'timeout 10 tcpdump -i eth1 icmp -w tcpdumptest_${logSuffix}.pcap &'"
+	ping_test $logSuffix
+}
+
+
+UTIL_FILE="./utils.sh"
+
+# Source utils.sh
+. ${UTIL_FILE} || {
+		echo "ERROR: unable to source ${UTIL_FILE}!"
+		echo "TestAborted" > state.txt
+		exit 0
+}
+
+# Source constants file and initialize most common variables
+UtilsInit
+
+# Script start from here
+LogMsg "*********INFO: Script Execution Started********"
+
+# check for client and server ips are present
+for ip in ${client} ${server}
+do
+	CheckIP ${ip}
+	if [ $? -eq 1 ]; then
+		LogErr "ERROR: Please provide valide client and server ip. Invalide ip: ${ip}"
+		SetTestStateAborted
+		exit 0
+	fi
+done
+
+if [ "${ACTION}" == "DROP" ];then
+	LogMsg "Initializing validation for XDP Action ${ACTION}"
+
+	# Ping test before changing action
+	ping_test "before"
+	check_exit_status "ping test before drop config"
+
+	# Build xdpdump with DROP Config
+	ssh ${client} "cd bpf-samples/xdpdump && make clean && export CFLAGS='-D __ACTION_DROP__ -I../libbpf/src/root/usr/include' && make"
+	check_exit_status "Building xdpdump with DROP config"
+
+	# Ping test after changing action
+	ping_test "after"
+	check_exit_status "ping test after drop config"
+
+	LogMsg "Starting analysis on number of ping packets drop.."
+	packetDropBefore=$( ssh ${server} "grep -oP '\d+(?=% packet loss)' ~/pingOut_before.txt" )
+	LogMsg "Packet drop before $packetDropBefore"
+	packetDropAfter=$( ssh ${server} "grep -oP '\d+(?=% packet loss)' ~/pingOut_after.txt" )
+	LogMsg "Packet drop after $packetDropAfter"
+	
+	if [ $((packetDropBefore + 30)) -lt $((packetDropAfter)) ];then
+		LogMsg "Test case executed successfully"
+		SetTestStateCompleted
+	else
+		LogErr "Ping packets are not dropped. Please check pingOut logs"
+		SetTestStateFailed
+	fi
+elif [ "${ACTION}" == "TX" ];then
+	LogMsg "Initializing validation for XDP Action ${ACTION}"
+
+	# Run ping from server and tcpdump with XDPDUmp on client
+	ping_with_tcpdump "before"
+	check_exit_status "tcpdump ping test before tx config"
+
+	LogMsg "Build patched XDpdump"
+	ssh ${client} "cd bpf-samples/xdpdump && make clean && export CFLAGS='-D __ACTION_TX__ -I../libbpf/src/root/usr/include' && make"
+	check_exit_status "Building xdpdump with TX config"
+
+	# Run Ping from server and tcpdump with xdpdump on client
+	# Get count of tcpdump and ping packets
+	ping_with_tcpdump "after"
+	check_exit_status "tcpdump ping test after tx config"
+
+	packetDropBefore=$( ssh ${server} "grep -oP '\d+(?=% packet loss)' ~/pingOut_before.txt" )
+	LogMsg "Packet drop before $packetDropBefore"
+	packetDropAfter=$( ssh ${server} "grep -oP '\d+(?=% packet loss)' ~/pingOut_after.txt" )
+	LogMsg "Packet drop after $packetDropAfter"
+
+	if [ $((packetDropAfter - packetDropBefore)) -le 10 ];then
```

Comment:
why 10?

Reply:
I have assumed max 10% drop due to the network.